.github/workflows/ci.yaml

```diff
@@ -83,44 +83,87 @@ jobs:
   tests:
     name: Run Dockerized Tests
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.9", "3.11" ]
     steps:
     - uses: actions/checkout@v4
+
+    - name: Set up python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: pip
+
     # TODO: re-use the next 2 actions instead of duplicating
     - name: Extract git branch
       id: git_info
       run: |
         echo "branch=$(echo ${GITHUB_BASE_REF#refs/heads/})" >> $GITHUB_OUTPUT
+
     - name: Resolve docker cache tag
       id: docker_cache
       run: |
         export version_suffix=$(echo "${{ steps.git_info.outputs.branch }}" | grep -E "^[0-9]+\.[0-9]+\.x$" | tr -d '.')
         export unstable_tag=$(if [ -z "$version_suffix" ]; then echo "unstable-cache"; else echo "unstable-cache-$version_suffix";fi)
         echo "tag=$(echo $unstable_tag)" >> $GITHUB_OUTPUT
+
+    - name: Install automation scripts dependencies
+      run: pip install -r automation/requirements.txt
+
     - name: Run Dockerized tests
-      run: MLRUN_DOCKER_REGISTRY=ghcr.io/ MLRUN_DOCKER_CACHE_FROM_TAG=${{ steps.docker_cache.outputs.tag }} make test-dockerized
+      run: |
+        MLRUN_DOCKER_REGISTRY=ghcr.io/ \
+        MLRUN_DOCKER_CACHE_FROM_TAG=${{ steps.docker_cache.outputs.tag }} \
+        MLRUN_PYTHON_VERSION=${{ matrix.python-version }} \
+        make test-dockerized
 
   integration-tests:
     name: Run Dockerized Integration Tests
     runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        # TODO: Ensure compatibility with Python 3.11 as well
+        # Currently, MLRun requires KFP with protobuf, which fails to install on Python 3.11.
+        python-version: [ 3.9 ]
```

Comment:
actually, I see that we sometimes use quotation and sometimes not (`"3.9"` vs `3.9`). Let's be aligned with that